tobago-example/tobago-example-demo/src/main/webapp/content/30-concept/53-collapsible/00-collapsible-box/Collapsible_Box.test.js

```diff
@@ -15,306 +15,206 @@
 
-  let TTT = new TobagoTestTool(assert);
-  TTT.action(function () {
-    show1Fn().dispatchEvent(new Event("click", {bubbles: true}));
-  });
-  TTT.waitForResponse();
-  TTT.asserts(2, function () {
-    assert.ok(content1Fn() !== null);
-    assert.equal(content2Fn().length, content2Length);
-  });
-  TTT.action(function () {
-    hide2Fn().dispatchEvent(new Event("click", {bubbles: true}));
-  });
-  TTT.waitForResponse();
-  TTT.asserts(2, function () {
-    assert.ok(content1Fn() !== null);
-    assert.equal(content2Fn(), null);
-  });
-  TTT.startTest();
+  let test = new JasmineTestTool(done);
+  test.do(() => show1Fn().dispatchEvent(new Event("click", {bubbles: true})));
+  test.wait(() => content1Fn() && content1Fn() !== null);
+  test.do(() => expect(content1Fn !== null).toBe(true));
+  test.do(() => expect(content2Fn().length).toBe(content2Length));
+  test.do(() => hide2Fn().dispatchEvent(new Event("click", {bubbles: true})));
+  test.wait(() => content1Fn() && content1Fn() !== null);
+  test.do(() => expect(content1Fn !== null).toBe(true));
+  test.do(() => expect(content2Fn()).toBe(null));
+  test.start();
 });
 
-QUnit.test("Full Server Request: close box 1, open box 2", function (assert) {
-  let hide1Fn = testFrameQuerySelectorFn("#page\\:mainForm\\:server\\:hide1");
-  let show2Fn = testFrameQuerySelectorFn("#page\\:mainForm\\:server\\:show2");
-  let content1Fn = testFrameQuerySelectorFn("#page\\:mainForm\\:server\\:content1");
-  let content2Fn = testFrameQuerySelectorFn("#page\\:mainForm\\:server\\:content2");
+it("Full Server Request: close box 1, open box 2", function (done) {
+  let hide1Fn = querySelectorFn("#page\\:mainForm\\:server\\:hide1");
+  let show2Fn = querySelectorFn("#page\\:mainForm\\:server\\:show2");
+  let content1Fn = querySelectorFn("#page\\:mainForm\\:server\\:content1");
+  let content2Fn = querySelectorFn("#page\\:mainForm\\:server\\:content2");
   let existContent2 = content2Fn() !== null;
 
-  let TTT = new TobagoTestTool(assert);
-  TTT.action(function () {
-    hide1Fn().dispatchEvent(new Event("click", {bubbles: true}));
-  });
-  TTT.waitForResponse();
-  TTT.asserts(2, function () {
-    assert.equal(content1Fn(), null);
-    assert.equal(content2Fn() !== null, existContent2);
-  });
-  TTT.action(function () {
-    show2Fn().dispatchEvent(new Event("click", {bubbles: true}));
-  });
-  TTT.waitForResponse();
-  TTT.asserts(2, function () {
-    assert.equal(content1Fn(), null);
-    assert.ok(content2Fn() !== null);
-  });
-  TTT.startTest();
+  let test = new JasmineTestTool(done);
+  test.do(() => hide1Fn().dispatchEvent(new Event("click", {bubbles: true})));
+  test.wait(() => content1Fn() === null);
+  test.do(() => expect(content1Fn()).toBe(null));
+  test.do(() => expect(content2Fn() !== null).toBe(existContent2));
+  test.do(() => show2Fn().dispatchEvent(new Event("click", {bubbles: true})));
+  test.wait(() => content1Fn() === null);
+  test.do(() => expect(content1Fn()).toBe(null));
```

Comment:
For cleaner code: You might use the same 'expect' as in line 104 and write: test.do(() => expect(content1Fn() !== null).toBe(false));